src/systems/elevator/utils/DoorTimer.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * \author Nick Lamprianidis <nlamprian@gmail.com>
+ * \date January 2021
+ */
+
+#include <memory>
+
+#include "DoorTimer.hh"
+
+namespace ignition
+{
+namespace gazebo
+{
+// Inline bracket to help doxygen filtering
+inline namespace IGNITION_GAZEBO_VERSION_NAMESPACE
+{
+namespace systems
+{
+class DoorTimerPrivate
+{
+  /// \brief Wait duration
+  public: std::chrono::steady_clock::duration waitDuration;
+
+  /// \brief Timeout time
+  public: std::chrono::steady_clock::duration timeoutTime;
+
+  /// \brief Function that gets called upon timeout
+  public: std::function<void()> timeoutCallback;
+
+  /// \brief Flag to indicate whether the doorway was blocked on the last update
+  /// time
```

Comment:
This information doesn't seem to be used outside of `Update`, do we need to keep it as a member variable?

Reply:
It's used to detect transitions between iterations, and I cannot set it as static in the method as there is more than one instance. Similarly, isActive goes into the same basket.

Reply:
It's used in Configure. It's re-initialized when the timer is started.